rsciio/utils/distributed.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2023 The HyperSpy developers
+#
+# This file is part of RosettaSciIO.
+#
+# RosettaSciIO is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# RosettaSciIO is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RosettaSciIO. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+
+import numpy as np
+import dask.array as da
+
+
+def get_chunk_slice(
+    shape,
+    chunks="auto",
+    block_size_limit=None,
+    dtype=None,
+):
+    """
+    Get chunk slices for the :func:`rsciio.utils.distributed.slice_memmap` function.
+
+    Takes a shape and chunks and returns a dask array of the slices to be used with the
+    :func:`rsciio.utils.distributed.slice_memmap` function. This is useful for loading data from a memmaped file in a
+    distributed manner.
+
+    Parameters
+    ----------
+    shape : tuple
+        Shape of the data.
+    chunks : tuple, optional
+        Chunk shape. The default is "auto".
+    block_size_limit : int, optional
+        Maximum size of a block in bytes. The default is None.
+    dtype : numpy.dtype, optional
+        Data type. The default is None.
+
+    Returns
+    -------
+    dask.array.Array
+        Dask array of the slices.
+    """
+
+    chunks = da.core.normalize_chunks(
+        chunks=chunks, shape=shape, limit=block_size_limit, dtype=dtype
+    )
+    chunks_shape = tuple([len(c) for c in chunks])
+    slices = np.empty(
+        shape=chunks_shape
+        + (
+            len(chunks_shape),
+            2,
+        ),
+        dtype=int,
+    )
+    for ind in np.ndindex(chunks_shape):
+        current_chunk = [chunk[i] for i, chunk in zip(ind, chunks)]
+        starts = [int(np.sum(chunk[:i])) for i, chunk in zip(ind, chunks)]
+        stops = [s + c for s, c in zip(starts, current_chunk)]
+        slices[ind] = [[start, stop] for start, stop in zip(starts, stops)]
+    return da.from_array(slices, chunks=(1,) * len(shape) + slices.shape[-2:]), chunks
+
+
+def slice_memmap(sl, file, dtypes, shape, **kwargs):
```

Comment:
Avoid using an abbreviation?
```suggestion
def slice_memmap(slices, file, dtypes, shape, **kwargs):
```

Reply:
Thats a good idea.  I've kind of gotten into the habit of using sl or slic just because ``slice`` is protected but slices is probably the better thing to use.